app/assets/javascripts/protocols/steps.js.erb

```diff
@@ -593,21 +593,15 @@
           animateSpinner(null, false);
           setupAssetsLoading();
         },
-        error: function(e) {
-          $form.after(xhr.responseJSON.html);
-          var $new_form = $form.next();
-          $form.remove();
-
-          $errInput = $new_form.find(".form-group.has-error")
-                               .first()
-                               .find("input");
-          renderFormError(e, $errInput);
-
-          formCallback($form);
-          applyCancelOnNew();
+        error: function(xhr) {
+          if (xhr.responseJSON['assets.file'] != null) {
```

Comment:
you don't need `!=  null` just negate the xhr.responseJSON['assets.file']
`if (! xhr.responseJSON['assets.file']) { .... }`